File: students/students/src/MON.SPA/src/models/studentClass/studentClassBaseModel.js
import Constants from '@/common/constants.js';

export class StudentClassBaseModel {
  constructor(obj = {},  moment = {}) {
    this.id = obj.id;
    this.personId = obj.personId;
    this.schoolYear = obj.schoolYear;
    this.classId = obj.classId;
    this.selectedClassTypeId = obj.selectedClassTypeId;
    this.enrollmentDate = (obj.enrollmentDate && moment) ? moment(obj.enrollmentDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.dischargeDate = (obj.dischargeDate && moment) ? moment(obj.dischargeDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.positionId = obj.positionId;
    this.profession = obj.profession;
    this.eduForm = obj.eduForm;
    this.classGroup = obj.classGroup;
  }
}

File: students/students/src/MON.Models/StudentModels/Class/StudentClassBaseModel.cs
namespace MON.Models
{
    using System;

    public class StudentClassBaseModel
    {
        public int? Id { get; set; }
        public int PersonId { get; set; }
        public short SchoolYear { get; set; }
        public int ClassId { get; set; }
        public int? SelectedClassTypeId { get; set; }
        public DateTime? EnrollmentDate { get; set; }
        public DateTime? EntryDate { get; set; }
        public DateTime? DischargeDate { get; set; }
    }
}
